src/garage/_dtypes.py

```diff
@@ -807,10 +849,26 @@ def __new__(cls, env_spec, observations, actions, rewards,
                     'dimension of '
                     'length {}, but got key {} with batch size {} instead.'.
                     format(inferred_batch_size, key, val.shape[0]))
+        
+        # episode_infos
+        for key, val in episode_infos.items():
+            if not isinstance(val, (dict, np.ndarray)):
+                raise ValueError(
+                    'Each entry in episode_infos must be a numpy array or '
+                    'dictionary, but got key {} with value type {} instead.'
+                    'instead'.format(key, type(val)))
+
+            if (isinstance(val, np.ndarray)
+                    and val.shape[0] != inferred_batch_size):
```

Comment:
this is correct in the case of TimeStepBatch (here), but not in the base of EpisodeBatch (above).